js/geocoder.js

```diff
@@ -1,65 +1,40 @@
-var geocoder = function () {
-    //var _queryName = '';
-    // return the query name from the result
-    function _setQueryName(location) {
-        var _queryName = '';
-        $.each(location, function (i, namePart) {
-            if (i === location.length - 1) {
-                _queryName = _queryName + ', <span>' + namePart + '</span>';
-            } else {
-                _queryName = _queryName + namePart + ' ';
-            }
-        });
-        return _queryName;
-    }
+var lat,
+    lon,
+    latID,
+    longID,
+    featureID;
 
-    // we need to show a few things that aren't in the feature
-    // the actual query ran and the attribution
-    function _setProperties(data) {
-        // single address
-        if (data.features === undefined) {
-            $.each(data, function(i, result) {
-                $.each(result.features, function(j, feature) {
-                    feature.properties.query = _setQueryName(result.query);
-                    feature.properties.attribution = result.attribution;
-                });
-            });
-        // batch
-        } else {
-            $.each(data.features, function(j, feature) {
-                feature.properties.query = _setQueryName(data.query);
-                feature.properties.attribution = data.attribution;
-            });
-        }
-        return data;
-    }
+function _setID(data) {
```

Comment:
I have a concern that lat/lon alone will not result in unique IDs. (imagine an apartment complex that is represented by its centroid). So for each individual address, we'd likely return the same point and thus they'll be given the same id by this code. At this point, it doesn't look like you're depending on the ids being unique, since they are used exclusively for  spatial stuff (add hover/selected marker to some point.. and if the points are at the same place, nothing will seem flat out *wrong*), but this will still have secondary effects, like highlighting *every* point at a given lat/lon when you mouse over a link. This will likely be visually discernible and have some negative performance impact. This is just a word of caution--if this is the desired/acceptable behavior, you might consider renaming from id to "location" or "latlon" etc. so it is made clear that these lat + lon is *not* necessarily unique... even if it will be in most cases.